src/test/scala/scala/xml/AttributeTest.scala

```diff
@@ -65,4 +65,97 @@ class AttributeTest {
     assertEquals("apple", xml \@ "bar")
   }
 
-}
\ No newline at end of file
+  @Test
+  def attributePathRootNoAttribute: Unit = {
+    val xml = <foo />
+    assertEquals(NodeSeq.Empty, xml \ "@bar")
+  }
+
+  @Test(expected=classOf[IllegalArgumentException])
+  def attributePathIllegalEmptyAttribute: Unit = {
+    val xml = <foo />
+    xml \ "@"
+  }
+
+  @Test
+  def attributePathRootWithOneAttribute: Unit = {
+    val xml = <foo bar="apple" />
+    assertEquals(Group(Text("apple")), xml \ "@bar")
+    // assertEquals(NodeSeq.fromSeq(Seq(Text("apple"))), xml \ "@bar")
```

Comment:
Good catch. Looks like I put that there on purpose.  It is what the test would be if we used the proposed change in issue 43.  Granted, I didn't leave these comments consistently.